lib/connection_pool.rb

```diff
@@ -44,6 +44,32 @@ def self.wrap(options, &block)
     Wrapper.new(options, &block)
   end
 
+  def self.after_fork
+    ObjectSpace.each_object(ConnectionPool) do |pool|
+      # We're on after fork, so we know all other threads are dead.
+      # All we need to do is to ensure the main thread doesn't have a
+      # checked out connection
+      pool.checkin(force: true)
```

Comment:
Isn't this code running in the child main thread? Any main thread connection would have been checked out by the parent process's main thread. Will that use the same thread locals?

Reply:
> Will that use the same thread locals?

yes, the main thread state is kept.

So this codepath is to handle code such as:

```ruby
pool.checkout
fork
```

or 

```ruby
pool.with do
  fork
end
```